adam/perception/high_level_semantics_situation_to_developmental_primitive_perception.py

```diff
@@ -876,13 +879,20 @@ def _instantiate_object_schema(
                 if self._generator.ontology.is_subtype_of(property_, COLOR)
             )
 
-            if sub_object_colors:
-                sub_object_color: Any = self._chooser.choice(sub_object_colors)
+            if sub_object.schema.ontology_node in sub_object_nodes_to_rgb:
```

Comment:
Would it suffice to handle this at the level of this for loop without needing to pass anything as a function argument?  that is, before the `for sub_object in schema.sub_objects` loop, gather the ontology node types of the sub-objects and assign one color per ontology node type, then use those colors within the loop?

Reply:
I think it might!